src/main/java/co/cask/hydrator/plugin/batch/S3ToRedshiftAction.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.batch;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import com.google.common.base.Strings;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+import javax.annotation.Nullable;
+
+/**
+ * S3ToRedshift Action Plugin - Loads the data from S3 into Redshift.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("S3ToRedshift")
+@Description("S3ToRedshift Action that will load the data from AWS S3 bucket into the AWS Redshift table.")
+public class S3ToRedshiftAction extends Action {
+  private final S3ToRedshiftConfig config;
+
+  public S3ToRedshiftAction(S3ToRedshiftConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    config.checkKeysAndRoleForConnection();
+  }
+
+  @Override
+  public void run(ActionContext actionContext) throws Exception {
+    config.checkKeysAndRoleForConnection();
+    Connection connection = null;
+    Statement statement = null;
+    try {
+      Class.forName("com.amazon.redshift.jdbc4.Driver");
+      //Open a connection and define properties.
+      Properties props = new Properties();
+      props.setProperty("user", config.masterUser);
+      props.setProperty("password", config.masterPassword);
+      connection = DriverManager.getConnection(config.clusterDbUrl, props);
+      statement = connection.createStatement();
+
+      String commandToExecute = buildCopyCommand();
+      statement.executeUpdate(commandToExecute);
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException(e);
```

Comment:
Can you be more descriptive in your error messages? In this case, can you say something like "Could not find Redshift JDBC Driver class."